app/assets/javascripts/save-editor-progress.es6

```diff
@@ -3,16 +3,37 @@
 
     function submitPracticeEditorSaveForm() {
         $('#practice-editor-save-button').on('click', () => {
-            if ($('#form')[0].checkValidity()) {
-                    $('#form').submit();
+            let $form = $('#form');
+            if ($form[0].checkValidity()) {
+                    $form.submit();
             } else {
-                $('#form')[0].reportValidity();
+                $form[0].reportValidity();
             }
         });
     }
 
+    function saveEditorProgressOnContinue() {
+        let currentEndpoint = window.location.href.split('/').pop();
+        if (currentEndpoint != 'adoptions') {
+            $('.continue-and-save').on('click', (event) => {
+                event.preventDefault();
+                let $form = $('#form');
+                let nextEndpoint = $(event.target).data('next');
+
+                $form.append(`<input type='hidden' name='next' value=${nextEndpoint}>`);
+
+                if ($form[0].checkValidity()) {
+                    $form.submit();
+                } else {
+                    $form[0].reportValidity();
+                }
+            });
+        }
+    }
+
```

Comment:
if we use the slug-hash method mentioned below, we probably don't need this anymore